### DISCUSSION TITLE
Home Assistant Devices not showing up in HC

### DISCUSSION THREAD

**Discussion by gerryvs** [alexander-vitishchenko/hc3-to-mqtt#48] [Q&A] [answered]
Dear Community,

I recently started to play arround with Fibaro and Home Assistant and since both have there advantage and disadvantages I was trying to integrate both. So I bounced into the HC3 <-> MQTT Bridge quickapp and have setup Mosquitto as MQTT broker.
All seem to be fine, I don't got any errors and I do see the Fibaro devices from HCL in my Home Assistant. The last step I was hoping to also see Home Assistant devices in HCL but this doesn't seem to be the case.
Is the intention of the Bridge only to make devices from HC available in Home Assistant or am I missing somewhere something in my setup ?

**Comment by alexander-vitishchenko** [comment]
Hi @gerryvs,

Have you configured your Home Assistant to use the newly installed MQTT broker? Home Assistant doesn't get connected to MQTT broker automatically, and you need to follow the steps described at https://www.home-assistant.io/integrations/mqtt

Can you please share your Fibaro logs starting from the line "------- HC3 <-> MQTT BRIDGE"?

Note: this integration is about making Fibaro HC3/HCL devices available from Home Assistant, not vice versa :-)

**Reply by gerryvs** [reply]
Hi Alexander,

Thanks for your fast answer ! And great work your doing ! 
I did install the broker on Home assistant, and I see it picks up the devices from the HCL, but the other way arround seems to be missing something. 

Picture on HomeAssistant side.
![image](https://user-images.githubusercontent.com/40800705/222963607-af251df1-5a14-42d2-85f4-f0017dca30c0.png)

Picture on HCL Side
![image](https://user-images.githubusercontent.com/40800705/222963846-b926eaab-a669-484e-a141-737474781f3b.png)

Config Quickapp
![image](https://user-images.githubusercontent.com/40800705/222963927-a2e4d856-2aca-48e7-a3d3-0121c718ce0a.png)

[HC3-log.txt](https://github.com/alexander-vitishchenko/hc3-to-mqtt/files/10891911/HC3-log.txt)

Thanks a lot for your response ! 
Gerry

**Comment by alexander-vitishchenko** [comment, marked as answer]
There are certainly errors in the logs, e.g. 'resolve: Host not found (authoritative)'. Are you sure MQTT URL value is correct at your QuickApp variables?

And I can also see five "HC3 <-> MQTT Bridge" entities at your Home Assistant screenshot. There should one normally. Can you please check your Fibaro instance and make sure there is only one QuickApp instance uploaded?

**Reply by gerryvs** [reply]
Hi Alexander,
I've reinstalled the the Quickapp a couple of times but deleted them as I noticed something was not as it should be. I've removed them also in HA now. so just one instance left there. 

The errors in the log probably are coming from the time I installed the app and did not yet configure the account and url.

Is it somehow possible or do you have any plans to also make devices from Home Assistant available in HC ?

Thanks a lot !
Gerry

**Comment by alexander-vitishchenko** [comment]
1) Can you please advise if you can now see your Fibaro HC devices at Home Assistant, or if the issue persists?

2) It is technically possible to make Home Assistant devices/entities available from Fibaro HC in the future, it just takes much time to get implemented.

**Reply by gerryvs** [reply]
Goodmorning Alexander, 

I can see the HC devices inside Home Assistant now and I also see them updating. Do you have an idea why I can not add the Aeotec Energy meter to the Energy control panel ? I see the values updating through QTT but can't select it for energy.

![image](https://user-images.githubusercontent.com/40800705/223064393-133c66d8-9772-46a5-8db7-31c034de0dce.png)


It would be nice in the future if it also could work the other way around, but having the devices now in Home Assistant simplifies already a lot. 

Thanks a lot for your great support !
Gerry

**Comment by alexander-vitishchenko** [comment]
1) Can you please send me the device JSON for those sensors that are shown with "eye" icon? They seem to be recognized as generic sensors, not energy and I can possibly adjust it easily.

2) As for managing Home Assistant devices/entities - this kind of work would require me to work like on a commercial project, as I can hardly make it during evenings or weekends. But maybe this will change in the future.

**Reply by gerryvs** [reply]
Hi Alexander,

Not sure how I can get JSON for a sensor, I've downloaded the diagnostic data. In HA they are referring to this article to share it with developer of an integration. 
https://www.home-assistant.io/more-info/statistics/


[mqtt-a52fdcb2744bb17cb03ffb98ff483f67-74-4f25eec4bb6211e65274a264a197b070.json.txt](https://github.com/alexander-vitishchenko/hc3-to-mqtt/files/10896676/mqtt-a52fdcb2744bb17cb03ffb98ff483f67-74-4f25eec4bb6211e65274a264a197b070.json.txt)

**Comment by alexander-vitishchenko** [comment]
This is how to get the JSON from Fibaro HC (copied from https://github.com/alexander-vitishchenko/hc3-to-mqtt)

![image](https://user-images.githubusercontent.com/1070777/223080722-14dbeec8-6174-454e-bdc2-2276fd8b1e22.png)

**Reply by gerryvs** [reply]
Hi Alexander,

The device is a aeotec Home Energy Monitor with 3 clamps, I'm using only 2 of them, 1 for the consumption from the grid and second one measures production from the solar panels.  The main node in my case is 74, will you also need screenshots for the subdevices ? 

![image](https://user-images.githubusercontent.com/40800705/223091839-121ba1c0-d176-4ba4-b762-3aacbbd55713.png)
![image](https://user-images.githubusercontent.com/40800705/223092008-2c710cbf-1fd2-4ea9-9bd9-f00e494e8c0d.png)
![image](https://user-images.githubusercontent.com/40800705/223092284-78b3394d-d325-4ce3-94da-8d9f35180e4c.png)

When I take the details from the Grid for example
![image](https://user-images.githubusercontent.com/40800705/223093556-8011bf9c-8a36-457e-90a3-606b89b8b7f4.png)
![image](https://user-images.githubusercontent.com/40800705/223093662-815d2bfa-d288-4a98-9b14-d3e54201639c.png)
![image](https://user-images.githubusercontent.com/40800705/223093753-d35c99e9-3e7f-45c6-8be4-30de579f2348.png)

This is how I see the energy panel in HC
![image](https://user-images.githubusercontent.com/40800705/223094112-1c6e8a01-e10d-4faa-bfb3-d89c806bc467.png)

and here are all values displayed by the device
![image](https://user-images.githubusercontent.com/40800705/223094817-db222c27-e55a-454d-a372-a8e543d08861.png)


[Device-75_response_1678099922138.txt](https://github.com/alexander-vitishchenko/hc3-to-mqtt/files/10897324/Device-75_response_1678099922138.txt)
[Device-74_response_1678100481742.txt](https://github.com/alexander-vitishchenko/hc3-to-mqtt/files/10897381/Device-74_response_1678100481742.txt)

[Device-83_response_1678100682886.txt](https://github.com/alexander-vitishchenko/hc3-to-mqtt/files/10897413/Device-83_response_1678100682886.txt)

**Comment by alexander-vitishchenko** [comment]
Hi @gerryvs,

Thank you for the inputs, this is good for the first iteration. Then I publish the new version of the QuickApp and we see if there is anything else we need to do

**Reply by gerryvs** [reply]
Great ! Thanks a lot and let me know if I need to perform some testing.

**Reply by alexander-vitishchenko** [reply]
Hi @gerryvs,

I've just released the new 1.0.221 version.

Can you please check if your Fibaro energy sensors are identified by Home Assistant properly now?

**Reply by gerryvs** [reply]
Hi @alexander-vitishchenko  

Looks I can select them now as energy monitors in Home Assistant ! Thanks a lot for the update ! 
I will monitor the coming days how they behave in HA, but at least I have the options to add them to the energy panel now.
